src/v/cluster/rm_stm.cc

```diff
@@ -1331,6 +1331,37 @@ model::offset rm_stm::last_stable_offset() {
         // transactions.
         lso = std::min(first_tx_start, next_to_apply);
     } else if (synced_leader) {
+        ////////////////  WARNING ///////////
+        // there is a real bug lurking here that overestimates the LSO beyond
+        // an open transaction.
+        //
+        // The issue here if a begin_tx is replicated but not yet applied
+        // in the stm and a caller queries LSO, we return LSO beyond the
+        // begin_tx which is incorrect. The problem is _active_tx_producers
```

Comment:
nit: Something is wrong with this sentence (or with my ability to read English). `The issue here IS THAT ...`?